[EdgeTPU] Update the testing codes to register EdgeTPUToolchain

rhqjatn2398 opened the pull request:
This draft modify the testing codes to test registration of EdgeTPUToolchain.
Also, This commit exports BackendMap interface to use it in the testing codes.

ONE-vscode-DCO-1.0-Signed-off-by: Bumsoo Ko <rhqjatn2398@naver.com>

rhqjatn2398 committed:
[EdgeTPU] Update the testing codes to register EdgeTPUToolchain

This commit modify the testing codes to test registration of EdgeTPUToolchain.
Also, This commit exports BackendMap interface to use it in the testing codes.

ONE-vscode-DCO-1.0-Signed-off-by: Bumsoo Ko <rhqjatn2398@naver.com>

rhqjatn2398 committed:
[EdgeTPU] Fix the testing codes

This commit fixes the 'registers a backend' testing codes.

After the calling of API.registerBackend, it adds the element to gToolchainEnvMap and globalBackendMap.
So globalBackendMap should be cleaned up.

ONE-vscode-DCO-1.0-Signed-off-by: Bumsoo Ko <rhqjatn2398@naver.com>

profornnan commented on src/Tests/Toolchain/ToolchainProvider.test.ts:
```diff
@@ -34,28 +34,22 @@ import {
   teardown(function () {
-    Object.keys(gToolchainEnvMap).forEach(
-      (key) => delete gToolchainEnvMap[key]
-    );
+    if (gToolchainEnvMap[backendName] !== undefined) {
+      delete gToolchainEnvMap[backendName];
+    }
   });
```
I think teardown function should be located at the bottom of "Toolchain" test suite :-D
(related to https://github.com/Samsung/ONE-vscode/pull/1686#discussion_r1336789547)

rhqjatn2398 replied:
Oh, Yes.
Thanks for reminding!

I'll fix the codes. 😃

profornnan commented on src/Tests/View/InstallQuickInput.test.ts:
```diff
@@ -46,30 +45,22 @@ suite("View", function () {
     teardown(function () {
-      Object.keys(gToolchainEnvMap).forEach(
-        (key) => delete gToolchainEnvMap[key]
-      );
+      if (gToolchainEnvMap[backendName] !== undefined) {
+        delete gToolchainEnvMap[backendName];
+      }
     });
```
Same here

rhqjatn2398 replied:
👍

rhqjatn2398 committed:
[EdgeTPU] Update the testing codes

This commit moves the teardown function to the bottom of the test suite.

ONE-vscode-DCO-1.0-Signed-off-by: Bumsoo Ko <rhqjatn2398@naver.com>

profornnan reviewed (approved):
LGTM 👍

rhqjatn2398 committed:
Merge branch 'fix/toolchain-test' into fix/toolchain-test-conserve

rhqjatn2398 commented:
These commits are cherry-picked on #81
